src/site/stages/build/plugins/check-collections.js

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable no-console */
+const path = require('path');
+const collections = require('../data/collections.json');
+
+/**
+ * This checks whether there are broken collections in a front-matter .md file.
+ * It checks whether a collection or children is available in the MetalSmith collections list
+ * @param {object} buildOptions
+ */
+
+const checkCollections = () => (files, metalsmith, done) => {
+  const VALID_COLLECTIONS = Object.keys(collections);
+
+  const brokenCollections = Object.keys(files)
+    .filter(
+      fileName =>
+        path.extname(fileName) === '.html' || path.extname(fileName) === '.md',
+    )
+    // returns an Array of Error messages for pages that have collection errors
+    .reduce((accum, key) => {
+      const file = files[key];
+
+      // Checks whether a page has a collection which is an Array. A page can have more then 1 collection
+      console.log(`Checking ${key}`);
+
+      if (file.collection) {
+        if (
+          file.collection.constructor.name === 'Array' &&
+          file.collection.length > 0
+        ) {
+          file.collection.forEach(collectionName => {
+            // Checks if the collection is valid
+            if (!VALID_COLLECTIONS.includes(collectionName)) {
+              accum.push(
+                `Error: Your [ collection "${collectionName}" ] does not exist in the collection: ${key}`,
+              );
+            }
+          });
+        } else if (file.collection.constructor.name === 'String') {
+          if (!VALID_COLLECTIONS.includes(file.collection)) {
+            accum.push(
+              `Error: Your [ collection "${
+                file.collection
+              }" ] does not exist in the collection: ${key}`,
+            );
+          }
+        }
+      }
+
+      // Checks if a page children is valid. A child collection is always going to be a string
+      if (file.children && !VALID_COLLECTIONS.includes(file.children)) {
+        accum.push(
+          `Error: Your [ children: "${
+            file.children
+          }" ] does not exist in the collection: ${key}`,
+        );
+      }
+
+      return accum;
+    }, []);
```

Comment:
I still don't think `.reduce` is the right choice of function here. When I think of `.reduce`, I think of iterating through an array to produce a single value, not another array. `.map` is what I think of for producing an array out of another.  You could also reduce some duplicate code by making sure `file.collection` is always the same data structure. To be a bit more expressive, you could use a `Set` for `VALID_COLLECTIONS`.

This is untested, but something like -

```js
.filter(
  fileName =>
    path.extname(fileName) === '.html' || path.extname(fileName) === '.md',
)

// Map the file names into an array of error messages, or null if there's no error.
.map(fileName => {
  let errorMessage = '';
  const file = files[fileName];

  if (file.collection) {
    const collectionIsString = file.collection.constructor.name === 'String';
    const collections  = collectionIsString ? [file.collection] : file.collection;
    
    collections.forEach(collectionName => {
      if (!VALID_COLLECTIONS.has(collectionName)) {
        errorMessage += `Error: Your [ collection "${collectionName}" ] does not exist in the collection: ${key}\n`;
      }
    });
  }

  // Checks if a page children is valid. A child collection is always going to be a string
  if (file.children && !VALID_COLLECTIONS.has(file.children)) {
    errorMessage += `Error: Your [ children: "${file.children}" ] does not exist in the collection: ${key}`;
  }

  return errorMessage;
})

// Remove null error messages
.filter(error => !!error);
```